### PR TITLE
Canon EOS R5/R6 changes from ART

### PR DESCRIPTION
**Pull request by kdm9**
This brings in one code fix and a bunch of DCP/camconst settings from ART. See #5882. Probably also solves #5901 

Cheers,
K

### CODE REVIEW

File: rtengine/canon_cr3_decoder.cc [hunk 1]
```diff
@@ -3133,6 +3133,7 @@ void DCraw::crxLoadRaw()
     }
 
 //  /*imgdata.color.*/maximum = (1 << hdr.nBits) - 1;
+  tiff_bps = hdr.nBits;
```
Comment (Thanatomanic) [inline rtengine/canon_cr3_decoder.cc]: Please fix the indentation here to make it consistent with the rest.

File: rtengine/camconst.json [hunk 2]
```diff
@@ -1149,6 +1149,22 @@ Camera constants:
         "raw_crop": [ 144, 46, 6744, 4500 ]
     },
 
+    { // Quality C
+        "make_model": "Canon EOS R5",
+        "dcraw_matrix" : [9766, -2953, -1254, -4276, 12116, 2433, -437, 1336, 5131],
+        "raw_crop" : [ 124, 92, 8220, 5486 ],
+        "masked_areas" : [ 94, 20, 5578, 122 ],
+        "ranges" : { "white" : 16383 }
```
Comment (Thanatomanic) [inline rtengine/camconst.json]: Based on my analysis [here](https://github.com/Beep6581/RawTherapee/issues/5901), I would recommend 16382 as a more appropriate white level.

**Comment by Thanatomanic** [comment]
@kdmurray91 Thank you for this PR. The changes to `camconst.json` are helpful, as is the fix for the bbp. Please check my minor comments.
The DCP files don't seem to be recognized automatically by RT, I think because the Exif cannot be read.

File: rtengine/camconst.json (hunk 2 above)
Comment (kdm9) [reply rtengine/camconst.json]: also changed for the R6

**Comment by kdm9** [comment]
Thanks for merging @Thanatomanic 

> The DCP files don't seem to be recognized automatically by RT, I think because the Exif cannot be read.

Is there anything I can do to fix this?

**Comment by Thanatomanic** [comment]
@kdmurray91 Not unless you want to code the support yourself into our custom `rtexif` metadata handling, or you want to simultaneously merge https://github.com/Beep6581/RawTherapee/pull/5889 and convince the maintainers of exiv2 to implement reading CR3 metadata. 😉  (some not-so-light reading here https://discuss.pixls.us/t/canon-cr3-raw-support-your-move/14192 - it gets progressively murkier)

**Comment by kdm9** [comment]
Any chance of taking the Digikam approach -- build off the existing CR3 parsing code and pass EXIF/XMP "chunks" into libexiv2 for metadata-level parsing? I'm happy to have a go at kluging something along those lines together if it means proper CR3/R5 support in RT.

Cheers,
K

**Pull request merged**